sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1902,202 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+def _get_cache_dict():
```

Comment:
You can freeze the defaultdict by setting `dd.default_factory = None`. Would doing that at the end of `fit` be an appropriate fix?